### ISSUE TITLE
Potential bug in complex exp() function

### ISSUE THREAD

**Issue by JorisDeRidder** [rust-num/num-complex#103]
The following code 
```
use num::complex::Complex;

fn main() {

    let z = Complex::<f64>::new(f64::NEG_INFINITY, f64::INFINITY);
    println!("z = {:?}", z);
    println!("exp(z) = {:?}", z.exp());
}
```
returns
```
z = Complex { re: -inf, im: inf }
exp(z) = Complex { re: NaN, im: NaN }
```
while its C++ equivalent 
```
#include <iostream>
#include <complex>

using namespace std;

int main() {

    complex<double> z(-INFINITY, INFINITY);
    cout << "z = " << z << endl;
    cout << "exp(z) = " << exp(z) << endl;

    return 0;
}
```
returns a different outcome:
```
z = (-inf,inf)
exp(z) = (0,0)
```

Might it be that the [special values](https://en.cppreference.com/w/cpp/numeric/complex/exp) are not properly implemented in `num`?

**Comment by cuviper** [comment]
You're right, it is not trying to handle "special" values at all, directly computing: _e_<sup>a + _i_ b</sup> = _e_<sup>a</sup> (cos(a) + _i_ sin(b))

**Comment by cuviper** [comment]
Fixed by #104.
